src/preamble.js

```diff
@@ -802,6 +802,78 @@ function createExportWrapper(name, fixedasm) {
 }
 #endif
 
+#if ABORT_ON_WASM_EXCEPTIONS
+// Instrument all the exported functions to:
+// - abort if an unhandled exception occurs
+// - throw an exception if someone tries to call them after the program has aborted
+// Set settings.ABORT_ON_WASM_EXCEPTIONS for more info.
+__ATINIT__.unshift(function () { 
+#if DISABLE_EXCEPTION_CATCHING != 1
+  // With exception catching enabled we only handle exceptions at the top level
+  // so that the exception mechanics can work uninterrupted at the inner level.
+  // Don't do this with exception catching disabled to save some code size.
```

Comment:
`DISABLE_EXCEPTION_CATCHING == 2` has a special meaning. is the `!= 1` intentional, to treat 0 the same as 2?

Reply:
I see, thanks, make ssense.